core/trino-server-rpm/src/main/resources/dist/etc/init.d/trino

```diff
@@ -29,34 +29,37 @@ SERVER_LOG_FILE=$(grep -Po "(?<=^node.server-log-file=).*" $NODE_PROPERTIES | ta
 LAUNCHER_LOG_FILE=$(grep -Po "(?<=^node.launcher-log-file=).*" $NODE_PROPERTIES | tail -n 1)
 CONFIGURATION=(--launcher-config /usr/lib/trino/bin/launcher.properties --data-dir "$DATA_DIR" --node-config "$NODE_PROPERTIES" --jvm-config /etc/trino/jvm.config --config /etc/trino/config.properties --launcher-log-file "${LAUNCHER_LOG_FILE:-/var/log/trino/launcher.log}" --server-log-file "${SERVER_LOG_FILE:-/var/log/trino/server.log}")
 
+set -a
 # shellcheck disable=1091
 source /etc/trino/env.sh
+set +a
 
 start() {
     echo "Starting ${SERVICE_NAME} "
     if [ -z "${JAVA_HOME}" ]; then
+        # shellcheck disable=2016
         echo 'Warning: No value found for $JAVA_HOME. Default Java will be used.' >&2
-        sudo -u $SERVICE_USER /usr/lib/trino/bin/launcher start "${CONFIGURATION[@]}"
+        sudo -u $SERVICE_USER -E /usr/lib/trino/bin/launcher start "${CONFIGURATION[@]}"
```

Comment:
i don't think it's safe.

The role of this script is to provide controlled execution environment for the service being run. With `sudo -E` we remove isolation from the caller's env, so eg things like PATH or LANG could affect the state of the process

Reply:
It seemed to me that `sudo` is only used here to run the launcher with a less privileged user. Adding `-E` here doesn't prevent any user from running the launcher with custom env vars directly. And when this script is executed by systemd, I'm not sure if there's a risk of injecting vars.

WDYT about converting this into a systemd unit file? There's https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=

Reply:
i am OK with systemd in general, provided it won't cause compatibility problems for users.

we should make sure @electrum @willmostly see it among others

Reply:
Refactored into an associative array, where only its keys and values are exported as variables.